aten/src/ATen/core/dispatch/Dispatcher.cpp

```diff
@@ -268,7 +268,6 @@ RegistrationHandleRAII Dispatcher::registerFallback(DispatchKey dispatchKey, Ker
   std::lock_guard<std::mutex> lock(mutex_);
 
   auto idx = getDispatchTableIndexForDispatchKey(dispatchKey);
-  TORCH_CHECK(idx >= 0 && static_cast<uint64_t>(idx) < backendFallbackKernels_.size(), "idx=", idx);
   TORCH_CHECK(
     !backendFallbackKernels_[idx].kernel.isValid(),
```

Comment:
I'm pretty sure this is where the segfault happens: `idx` can be -1 on mobile builds, and we try to index into `backendFallbackKernels_`.